scripts/install_deps.sh

```diff
@@ -105,6 +105,9 @@ case $(uname -s) in
         brew update
         brew install boost
         brew install cmake
+        brew install cln
+        brew install z3
+        brew install cvc4
```

Comment:
These don't seem to be related to the PR?

Reply:
Well, they are needed for installation on macos. Would you prefer a separate PR?

Reply:
They are generally optional dependencies, it should be possible to compile without them. Does that not work there?

Reply:
It complained at least that cln was missing. I have the other two installed already. Could it be that cln is needed if z3 or cvc4 are present?

Reply:
Exactly, cln is needed by cvc4

Reply:
Things seem a little inconsistent across this script. `cvc4` is installed for Arch linux, but no other distro. `z3` installed for Debian and Ubuntu.

Reply:
Right, this is actually wrong I think

Reply:
So you prefer these lines left out?

Reply:
Yep